plugin/trino-querylog/src/main/java/io/trino/plugin/querylog/QuerylogListenerFactory.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.querylog;
+
+import io.trino.spi.eventlistener.EventListener;
+import io.trino.spi.eventlistener.EventListenerFactory;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+import java.io.File;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.security.KeyStore;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class QuerylogListenerFactory
+        implements EventListenerFactory
+{
+    public static final String QUERYLOG_SEND_CREATED = "trino.querylog.log.created";
+    public static final String QUERYLOG_SEND_COMPLETED = "trino.querylog.log.completed";
+    public static final String QUERYLOG_SEND_SPLIT = "trino.querylog.log.split";
+    public static final String QUERYLOG_CONNECT_INET_ADDRESS = "trino.querylog.connect.inet.address";
+    public static final String QUERYLOG_CONNECT_SCHEME = "trino.querylog.connect.scheme";
+    public static final String QUERYLOG_CONNECT_PORT = "trino.querylog.connect.port";
+    public static final String QUERYLOG_CONNECT_HEADERS_PREFIX = "trino.querylog.connect.headers.";
+    public static final String QUERYLOG_CONNECT_SSL_CERT_STORE_PATH = "trino.querylog.connect.sll.cert.store.path";
+    public static final String QUERYLOG_CONNECT_SSL_CERT_STORE_PASSWORD = "trino.querylog.connect.sll.cert.store.password";
```

Comment:
Please do handle configuration using annotation based Config class like we do in other places in Trino.
Easiest way to do that is to use `Bootstrap` here to obtain injector to get event listener instance:

```java
    @Override
    public EventListener create(Map<String, String> config)
    {
        Bootstrap app = new Bootstrap(
                binder -> binder.bind(QuerylogListener.class).in(Scopes.SINGLETON));

        Injector injector = app
                .doNotInitializeLogging()
                .setRequiredConfigurationProperties(config)
                .initialize();
        return injector.getInstance(QuerylogListener.class);
    }
```

Then you can create `QueryLogListenerConfig` class and inject instance to `QuerylogListener` via constructor parameter.